src/buffer/out/textBuffer.cpp

```diff
@@ -549,7 +549,11 @@ bool TextBuffer::IncrementCircularBuffer()
     _renderTarget.TriggerCircling();
 
     // First, clean out the old "first row" as it will become the "last row" of the buffer after the circle is performed.
-    const bool fSuccess = _storage.at(_firstRow).Reset(_currentAttributes);
+    // The VT standard requires that the new row is initialized with
+    // the current background color, but with no meta attributes set.
+    auto fillAttributes = _currentAttributes;
+    fillAttributes.SetMetaAttributes(0);

```

Comment:
Concerned about this one, actually -- this is used for traditional Windows console apps as well, and I don't know what the expected behavior is for new rows _outside_ of global VT processing mode.

Reply:
Ah yes. I should have mentioned I did actually have a look into that. I wrote a little test app that set the attributes with `SetConsoleTextAttribute` (using `COMMON_LVB_UNDERSCORE` and `COMMON_LVB_REVERSE_VIDEO`) and then output some text that would cause the screen to scroll.

In both the legacy console and the v2 console, a line feed doesn't actually fill the new row with the current attributes most of the time - only when it reaches the end of the buffer. That's just plain weird, but it seems that's the way it has always been, and that hasn't changed.

In the legacy console the underscore and reverse attributes don't seem to have any effect anywhere, so naturally that means the new rows filled when scrolling past the end of the buffer will also not have an underscore or have their attributes reversed. So in some ways this update more closely matches that legacy behaviour. It is a change from the current v2 behaviour though.

The bottom line is I'd be very surprised if anyone is relying on this behaviour, considering how strange and inconsistent it is. However, I know backward compatibility is important to you, so if you think it's a concern, I can probably add a check that only resets the attributes when in vt mode or something of the sort.

Reply:
I've now made it so this only applies in the specific case of the line feed scrolling when in vt mode. There are probably more situations where it should be applied, but for now this is the most conservative fix for this particular issue.